server/src/main/java/io/crate/analyze/TableIdentsExtractor.java

```diff
@@ -73,110 +67,115 @@ public static Collection<RelationName> extract(Iterable<? extends Symbol> symbol
+    private static class TableIdentSymbolVisitor extends DefaultTraversalSymbolVisitor<Collection<RelationName>, Void> {
 
         @Override
-        protected Collection<RelationName> visitSymbol(Symbol symbol, Void context) {
+        protected Void visitSymbol(Symbol symbol, Collection<RelationName> context) {
             throw new IllegalStateException(String.format(Locale.ENGLISH,
                 "Symbol '%s' not supported", Symbol.class.getName()));
         }
 
         @Override
-        public Collection<RelationName> visitAggregation(Aggregation symbol, Void context) {
-            return extract(symbol.inputs());
+        public Void visitAggregation(Aggregation symbol, Collection<RelationName> context) {
+            context.addAll(extract(symbol.inputs()));
+            return null;
         }
 
         @Override
-        public Collection<RelationName> visitReference(Reference symbol, Void context) {
-            return Collections.singletonList(symbol.ident().tableIdent());
+        public Void visitReference(Reference symbol, Collection<RelationName> context) {
+            context.add(symbol.ident().tableIdent());
+            return null;
         }
 
         @Override
-        public Collection<RelationName> visitDynamicReference(DynamicReference symbol, Void context) {
+        public Void visitDynamicReference(DynamicReference symbol, Collection<RelationName> context) {
             return visitReference(symbol, context);
         }
 
         @Override
-        public Collection<RelationName> visitFunction(Function symbol, Void context) {
-            return Collections.emptyList();
+        public Void visitFunction(Function symbol, Collection<RelationName> context) {
+            return null;
         }
 
         @Override
-        public Collection<RelationName> visitLiteral(Literal symbol, Void context) {
-            return Collections.emptyList();
+        public Void visitLiteral(Literal<?> symbol, Collection<RelationName> context) {
+            return null;
         }
 
         @Override
-        public Collection<RelationName> visitInputColumn(InputColumn inputColumn, Void context) {
-            return Collections.emptyList();
+        public Void visitInputColumn(InputColumn inputColumn, Collection<RelationName> context) {
+            return null;
         }
 
         @Override
-        public Collection<RelationName> visitField(ScopedSymbol field, Void context) {
-            return List.of(field.relation());
+        public Void visitField(ScopedSymbol field, Collection<RelationName> context) {
+            context.add(field.relation());
+            return null;
         }
 
         @Override
-        public Collection<RelationName> visitMatchPredicate(MatchPredicate matchPredicate, Void context) {
-            Set<RelationName> relationNames = new HashSet<>();
+        public Void visitMatchPredicate(MatchPredicate matchPredicate, Collection<RelationName> context) {
```

Comment:
I think this matches the default implementation and could be removed

Reply:
Thank you, updated it.